heap.c

```diff
@@ -100,6 +110,20 @@ siftup(struct heap *h, size_t pos)
 }
 
 void
+heap_heapify(struct heap *h)
+{
+	for (ssize_t i=ptrvec_size(h->vec)/2-1; i>=0; i--) {
+		siftup(h, i);
```

Comment:
we don't check it anywhere else in heap.c and we would have to change all of the function signatures to return the error code (they are void).  do you want to do this?